File: Unit_Testing_C++/src/product.cpp
// product.cpp
#include "product.h"

// Implement default constructor
Product::Product() {
    // Initialize default values or leave empty
}

Product::Product(int product_id, std::string name, double price, int stock)
    : product_id(product_id), name(name), price(price), stock(stock) {}

std::string Product::get_name() const {
    return name;
}

double Product::get_price() const {
    return price;
}


int Product::get_stock() {
    return stock;
}

int Product::get_product_id() const {
    return product_id;
}

void Product::update_stock(int quantity) {
    this->stock -= quantity;
}

File: Unit_Testing_Python/src/product.py
# product.py
class Product:
    def __init__(self, product_id, name, price, quantity):
        self.product_id = product_id
        self.name = name
        self.price = price
        self.quantity = quantity
        self.stock = quantity

    def display_info(self):
        return f"{self.name} - Price: ${self.price}, Quantity: {self.quantity}"

    def update_quantity(self, quantity):
        self.quantity += quantity
        self.stock += quantity
